forward-proxy/src/proxy.rs

```diff
@@ -61,20 +61,45 @@ impl ProxyHttp for ForwardProxy {
         //
         // Code below is for step 4(this is a client to RP), presenting the client's TLS certificate.
 
-        let addr = ctx.get(&consts::CtxKeys::UpstreamAddress.to_string()).unwrap();
+        let addrs = ctx.get(&consts::CtxKeys::UpstreamAddress.to_string()).unwrap();
```

Comment:
One day (when you're bored and i'm behind?), you can find and appropriately handle all  `.unwrap()` calls.